pennylane/templates/embeddings.py

```diff
@@ -0,0 +1,266 @@
+# Copyright 2018 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Embeddings
+==========
+
+**Module name:** :mod:`pennylane.templates.embeddings`
+
+.. currentmodule:: pennylane.templates.embeddings
+
+This module provides quantum circuit architectures that can embed features into a quantum state.
+The features are associated with gate parameters (see also Schuld & Killoran 2019
+:cite:`schuld2018quantum`).
+
+Qubit architectures
+-------------------
+
+Angle embedding
+***************
+
+Encodes ``n`` features into the rotation angles of ``n`` qubits. The rotation can be Pauli-X, -Y or -Z.
+
+.. autosummary::
+
+    AngleEmbedding
+
+Amplitude embedding
+*******************
+
+Encodes :math:`2^n` features into the amplitude vector of :math:`n` qubits. The absolute square of all features
+has to add up to one.
+
+.. note::
+
+    This embedding is a wrapper for PennyLane`s :mod:`QubitStateVector()`, and only works with backends which
```

Comment:
The QubitStateVector state prep method already requires a ket vector of size 2**len(wires) and prepares it in the fock basis. Dont we need another method to first prepare this ket from the given feature data? also why would a general amplitude encoding need to be done in the fock basis?

Reply:
Actually, these embeddings are a wrapper for the QubitStateVector method. This method can be seen as simple state preparation, but when fed a feature vector it becomes conceptually an embedding...

PS: AmplitudeEmbedding is supposed to work with qubit architectures, I have actually never thought of it preparing fock states...